src/aap_eda/core/management/commands/purge_log_records.py

```diff
@@ -0,0 +1,160 @@
+
+    def purge_log_records(self, ids: list[int], cutoff_date: str) -> None:
+        new_instance_logs = []
+        instances = models.RulebookProcess.objects.filter(
+            activation_id__in=ids,
+        )
+
+        for instance in instances:
+            new_instance_logs.append(
+                self.clean_instance_logs(instance, cutoff_date)
+            )
+
+        new_instance_logs = [
+            item for item in new_instance_logs if item is not None
+        ]
+        if len(new_instance_logs) > 0:
+            models.RulebookProcessLog.objects.bulk_create(new_instance_logs)
+
+    def clean_instance_logs(
+        self,
+        instance: models.RulebookProcess,
+        cutoff_date: str,
+    ) -> Union[models.RulebookProcessLog, None]:
+        try:
+            ts = datetime.strptime(cutoff_date, "%Y-%m-%d")
+        except ValueError:
+            raise ValueError("Timestamp must be in 'YYYY-MM-DD' format.")
+
+        log_records = models.RulebookProcessLog.objects.filter(
+            activation_instance=instance,
+            log_timestamp__lt=int(ts.timestamp()),
+        )
+        if not log_records.exists():
+            return
+
+        log_records.delete()
+        dt = f"{datetime.now().strftime('%Y-%m-%d %H:%M:%S')}"
+
+        return models.RulebookProcessLog(
+            log=(
+                f"All log records older than {cutoff_date} are purged at {dt}."
+            ),
+            activation_instance_id=instance.id,
+            log_timestamp=int(ts.timestamp()),
+        )
+
+    def get_activation_ids(
+        self, identifiers: List[Union[str, int]]
+    ) -> List[int]:
+        if all(isinstance(i, str) for i in identifiers):
+            ids = models.Activation.objects.filter(
+                name__in=identifiers,
+            ).values_list("id", flat=True)
+        elif all(isinstance(i, int) for i in identifiers):
+            ids = models.Activation.objects.filter(
+                id__in=identifiers,
+            ).values_list("id", flat=True)
+        else:
+            raise ValueError(
+                "Identifiers list must contain either all names (strings) or "
+                "all primary keys (integers)."
+            )
+
+        return list(ids)
+
+    @transaction.atomic
+    def handle(self, *args, **options):
+        input_ids = [int(id) for id in options.get("activation-ids") or []]
+        input_names = options.get("activation-names") or []
+        cutoff_date = options.get("date")
+
+        try:
+            if not bool(input_ids) and not bool(input_names):
+                activation_ids = [
+                    activation.id
+                    for activation in models.Activation.objects.all()
+                ]
+            else:
+                activation_ids = list(
+                    set(
+                        self.get_activation_ids(input_ids)
+                        + self.get_activation_ids(input_names)
+                    )
+                )
```

Comment:
Do we need this logic and the function `get_activation_ids`?
I believe we can just write a query that returns activations by both, ids or names, something like:
```
models.Activation.objects.filter(
    Q(id__in=ids) | Q(name=name)
).distinct()
```

Also, why do we have to run a query to get the ids and later run another query to get the rulebook_processes?I think believe we can just pass the queryse, then the ORM will run only one query.